src/types/presets/aws-amplify.ts

```diff
@@ -0,0 +1,161 @@
+   * Leave empty for only allowing the deployment domain to use Image Optimization.
+   */
+  domains: string[];
+
+  /**
+   * Array of allowed external patterns that can use Image Optimization.
+   * Similar to `domains` but provides more control with RegExp.
+   */
+  remotePatterns: {
+    /** The protocol of the allowed remote pattern. Can be `http` or `https`. */
+    protocol?: "http" | "https";
+    /**
+     * The hostname of the allowed remote pattern.
+     * Can be literal or wildcard. Single `*` matches a single subdomain.
+     *  Double `**` matches any number of subdomains.
+     * We will disallow blanket wildcards of `**` with nothing else.
+     */
+    hostname: string;
+    /** The port of the allowed remote pattern. */
+    port?: string;
+    /** The pathname of the allowed remote pattern. */
+    pathname?: string;
+  }[];
+
+  /** Array of allowed output image formats. */
+  formats?: (
+    | "image/avif"
+    | "image/webp"
+    | "image/gif"
+    | "image/png"
+    | "image/jpeg"
+  )[];
+
+  /** Cache duration (in seconds) for the optimized images. */
+  minimumCacheTTL?: number;
+
+  /** Allow SVG input image URLs. This is disabled by default for security purposes. */
+  dangerouslyAllowSVG?: boolean;
+};
+
+export interface AmplifyDeployManifest {
+  /** The `version` property dictates the version of the Deployment Specification that has been implemented */
+  version: 1;
+
+  /**
+   * The routes property allows framework authors to leverage the routing rules primitive.
+   * Routing rules provide a mechanism by which incoming request paths can be routed to a specific target
+   * in the deployment bundle. Routing rules only dictate the destination of an incoming request and they are
+   * applied after rewrite/redirect rules have already transformed the request.
+   *
+   * Limits for routing rules:
+   * - A limit of 25 rules will be enforced on the routes array
+   */
+  routes?: AmplifyRoute[];
+
+  /**
+   * The imageSettings property allows framework authors to customize the behavior of the image optimization primitive,
+   * which provides on-demand optimization of images at runtime.
+   */
+  imageSettings?: AmplifyImageSettings;
+  /**
+   * Metadata about the provisioned compute resource(s). Each item in the array is an object that contains metadata
+   * about that compute resource.
+   *
+   * For example, given the following directory structure:
+   * ```
+   * .amplify
+   * └── compute
+   *     └── default
+   *         └── index.js
+   * ```
+   * The `computeResources` property would look like:
+   * ```
+   * [
+   *   {
+   *     name: 'default',
+   *     runtime: 'nodejs16.x',
+   *     entrypoint: 'index.js',
+   *   }
+   * ]
+   * ```
+   */
+  computeResources?: AmplifyComputeConfig[];
+
+  // Framework Metadata
+  framework: {
+    name: string;
+    version: string;
+  };
+}
+
+export interface AWSAmplifyOptions {
+  computeStaticFallback?: boolean;
```

Comment:
Suggestion for this field name: `catchAllStaticFallback` ? My reasoning is that this flag is meant only for the catch-all (`/*`) route.

Also, you could have `Compute` to `Static` fallback elsewhere and that's totally valid (For example, wouldn't impact 404 pages):
```
{
  "path": "/foo/bar/*",
  "target": {
    "kind": "Compute",
    "src": "default"
  },
  "fallback": {
    "kind": "Static" --> This fallback is not problematic because it won't impact 404 pages
  }
},
{
  "path": "/*",
  "target": {
    "kind": "Compute", --> Adding fallback here will impact 404 pages
    "src": "default"
  }
}
```